src/ui/components/Library/useFilters.ts

```diff
@@ -25,6 +25,40 @@ export const LibraryContext = createContext<LibraryContextType>({
   preview: null,
 });
 
+export function encodeFilter(str: string) {
+  let encodedStr = "";
+  while (str.length > 0) {
+    const [match] = str.match(/^(.*?:)/) || [];
+    if (match) {
+      encodedStr += match;
+      str = str.substring(match.length);
+
+      const nextChar = str.substring(0, 1);
+      if (["'", '"', "["].includes(nextChar)) {
+        const closingChar = nextChar == "[" ? "]" : nextChar;
+        const [valueMatch] = str.substring(1).match(new RegExp(`^(.*?${closingChar})`)) || [];
+
+        if (valueMatch) {
+          encodedStr += encodeURIComponent(`${nextChar}${valueMatch}`);
+          str = str.substring(valueMatch.length + 1);
+        } else {
+          encodedStr += str.substring(1);
+          str = "";
+        }
+      } else {
+        const [nextWord] = str.match(/^(\w+)/) || [];
```

Comment:
are you sure you want a word boundary here? For instance, is `this-token` a valid next word? What about `a_wild_🦄_appears`?

I would be tempted to instead add a `BOUNDARY_TOKENS` const for whatever you consider the things that delimit a "word" and then do a `peekNextWord` similar to `peekUntil` above.